ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/KtlintSuppressionRuleTest.kt

```diff
@@ -0,0 +1,798 @@
+import org.junit.jupiter.params.provider.ValueSource
+
+class KtlintSuppressionRuleTest {
+    private val ktlintSuppressionRuleAssertThat =
+        KtLintAssertThat.assertThatRule { KtlintSuppressionRule() }
+
+    @Nested
+    inner class `Given a suppression annotation missing the rule set id prefix` {
+        @Test
+        fun `Given a @file Suppress annotation`() {
+            val code =
+                """
+                @file:Suppress("ktlint:bar", "ktlint:standard:foo", "ktlint:foo-bar")
+                """.trimIndent()
+            val formattedCode =
+                """
+                @file:Suppress("ktlint:standard:bar", "ktlint:standard:foo", "ktlint:standard:foo-bar")
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolations(
+                    LintViolation(1, 24, "Identifier to suppress ktlint rule must be fully qualified with the rule set id"),
+                    LintViolation(1, 61, "Identifier to suppress ktlint rule must be fully qualified with the rule set id"),
+                ).isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given a @file SuppressWarnings annotation`() {
+            val code =
+                """
+                @file:SuppressWarnings("ktlint:bar", "ktlint:standard:foo", "ktlint:foo-bar")
+                """.trimIndent()
+            val formattedCode =
+                """
+                @file:SuppressWarnings("ktlint:standard:bar", "ktlint:standard:foo", "ktlint:standard:foo-bar")
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolations(
+                    LintViolation(1, 32, "Identifier to suppress ktlint rule must be fully qualified with the rule set id"),
+                    LintViolation(1, 69, "Identifier to suppress ktlint rule must be fully qualified with the rule set id"),
+                ).isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given a Suppress annotation on a declaration`() {
+            val code =
+                """
+                @Suppress("ktlint:bar", "ktlint:standard:foo", "ktlint:foo-bar")
+                val foo = "foo"
+                """.trimIndent()
+            val formattedCode =
+                """
+                @Suppress("ktlint:standard:bar", "ktlint:standard:foo", "ktlint:standard:foo-bar")
+                val foo = "foo"
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolations(
+                    LintViolation(1, 19, "Identifier to suppress ktlint rule must be fully qualified with the rule set id"),
+                    LintViolation(1, 56, "Identifier to suppress ktlint rule must be fully qualified with the rule set id"),
+                ).isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given a SuppressWarnings annotation on a declaration`() {
```

Comment:
Should we add TC around
```
fun foo(
    a: Int /* ktlint-supression */
    b: Int // ktlint-suppression
)
```

Reply:
> > @atulgpt I you also want to contribute by reviewing changes, you're welcome to review this PR. You can review it in detail or just briefly. Normally, I would just merge the change as there are almost no regular contributors on the project that would like to do this. I you don't want to review, just let me know.
> > REVIEW suggestion:
> > ```
> > * The functional change for is contained in [this commit](https://github.com/pinterest/ktlint/pull/2068/commits/428d52e637d931754b3756f8912f64cea884f8d5)
> > 
> > * Other commits are supportive to the change
> > ```
> 
> Hi, @paul-dingemans I have added a few more TC suggestions. Will take a look in the logic as well

Wow, I am really amazed at the good quality of the review comments. Tnx!  I have left some review comments unanswered. I have to think about them.

Reply:
Good find. With latest change, the slightly rewritten example below:
```
fun foobar(
    foo: Int, /* ktlint-disable rule-foo */
    bar: Int, // ktlint-disable rule-bar
)
```
is changed to:
```
@Suppress("ktlint:standard:rule-bar", "ktlint:standard:rule-foo")
fun foobar(
    foo: Int,
    bar: Int,
)
```
which is not correct. It should have been:
```
@file:Suppress("ktlint:standard:rule-foo")

@Suppress("ktlint:standard:rule-bar")
fun foobar(
    foo: Int,
    bar: Int,
)
```

But I am doubting whether it is worth the effort to defend against this. A more realistic and troublesome example is:
```
fun foo() {
    /* ktlint-disable max-line-length */
}

fun bar() {
   val someLonnnnnnnnnnnnnnnnnnnnnnnnnnnnngIdentifierExceedingMaxLineLength = "bar"
}
```
The ktlint-disable directive does not have a logical scope (e.g. not bound to an ASTNode) but just to line number. I think that in such case, autocorrect is going to be difficult. Assuming that devs won't do stuff like this, I propose to do following:
* if a matching ktlint-enable directive in a block comment is found at a sibling node of the ktlint-disable directive (also in a block comment) then it is propagate to the first parent declaration/expression in same way as a disable directive in an EOL comment.
* Otherwise the ktlint-disable directive is just reported as an error which can not be autocorrected.

Example 1:
```
fun foobar(
     /* ktlint-disable rule-foo */
    foo: Int,
    bar: Int, // ktlint-disable rule-bar
    /* ktlint-enable rule-foo */
) {
    // do something
}
```
will be changed to:
```
@Suppress("ktlint:standard:rule-bar", "ktlint:standard:rule-foo")
fun foobar(
    foo: Int,
    bar: Int,
) {
    // do something
}
```

Example 2:
```
fun foobar(
     /* ktlint-disable rule-foo */
    foo: Int,
    bar: Int, // ktlint-disable rule-bar
) {
    /* ktlint-enable rule-foo */
    // do something
}
```
will result in a violation error which can not be autocorrected, and the code being changed to
```
@Suppress("ktlint:standard:rule-bar")
fun foobar(
     /* ktlint-disable rule-foo */
    foo: Int,
    bar: Int,
) {
    // do something
}
```

Reply:
Make sense. As no one would be using out-of-place enable and disable directives. Let me know once you update the final code. I would like to take a look and obviously will gaining lots of understanding around these